lms/services/dashboard.py

```diff
@@ -34,11 +34,8 @@ def __init__(  # noqa: PLR0913
         self._organization_service = organization_service
         self._h_authority = h_authority
 
-    def get_request_assignment(self, request) -> Assignment:
+    def get_request_assignment(self, request, assigment_id: int) -> Assignment:
```

Comment:
Taking the ID directly in this.

I've a commit on this PR that made a third possible location and these methods tried them in succession.

Take the ID directly, the view know where to get it from and will make refactoring this service easier in the future.

Reply:
This is how that change looks: https://github.com/hypothesis/lms/pull/6862/commits/4faab9ea85223fbec116cbe18d0615068fa44106